coalib/results/TextPosition.py

```diff
@@ -71,11 +70,70 @@ def __ge__(self, other):
         """
         if self.line is None or other.line is None:
             return True
+
         if self.line == other.line:
-            return (True
-                    if self.column is None or
-                    other.column is None
-                    else
-                    self.column >= other.column)
+            if self.column is None or other.column is None:
+                return True
+            else:
+                return self.column >= other.column
         else:
             return self.line > other.line
+
+    def __lt__(self, other):
+        """
+        Test whether ``self`` is behind the other
+        ``TextPosition``.
+
+        If the column in a ``TextPosition`` is ``None``, consider
+        whole line. If the line in a ``TextPosition`` is ``None``,
+        consider whole file.
+
+        :param other: ``TextPosition`` to compare with.
+        :return:      Whether this ``TextPosition`` is behind the other
+                      one.
+        """
+        if self.line is None or other.line is None:
```

Comment:
because of the possibilities of `None` values, we discussed the meaning of `None` fields:
- `column = None` -> Means any position in line
- `line = None` -> Means any position in file

So a `TextPosition` is not precise any more, it can mark many possible positions, it nearly indicates a range. Like @aptrishu said, because those `None`-ranges might span over the `TextPosition` to compare, we agreed on the current behaviour. This means in this case that for example `A < B` is not the same as `not A >= B`. The same applies for similar logical constructs.

Let's consider two `TextPositions` A and B:
- We consider A to be lower than B, if all marked positions by A are ahead of B and not touching each other. The same principle applies if A is greater than B.
- We consider A to be lower or equal than B, if nearly all positions of A are ahead of B. The remaining part is only allowed to be past B, if it was caused by a `None` value.
- We consider A to be equal to B, if they exactly match.

I think the crucial part is 2., and I can agree on another reasonable behaviour^^